### PR TITLE
Blaze: Show feedback banner and survey after campaign creation

### LINKED ISSUE #14036: Display survey to gather feedback on user experience of the campaign creation flow
- Ask for feedback from the Blaze success bottom sheet. In order to show the CTA requesting feedback the user must be creating at least their second campaign (not necessarily from the app though, we just want to ensure we ask for feedback when users had had the chance to experience how a Blaze campaign performs on their site).
- We’ll display a CTA like the one we show in AI product creation. The text could be How was the experience with Blaze? :thumbup: :-1: . This CTA will be placed below the bottom sheet’s Done button.

### PR DESCRIPTION
**Pull request by itsmeichigo**
Closes: #14036 

## Description

This PR adds a banner for user feedback on the success bottom sheet after Blaze campaign creation completes. This banner is displayed only when the user has created at least one campaign in the past. If they downvote on the feedback, a survey page is displayed for more detailed response.

Internal discussion: pe5sF9-3di-p2

Since the changes are on the UI, it's not straightforward to write tests. Any suggestions are welcome.

## Steps to reproduce

- Log in to a store eligible for Blaze without any campaign created in the past.
- Create a campaign for the store from the dashboard screen or Blaze campaign list.
- After the creation completes for the first campaign, confirm that the feedback banner is not displayed on the success bottom sheet.
- Create another campaign and confirm that this time the feedback banner is displayed.
- Tap on 👍 on the feedback banner, confirm that `blaze_campaign_creation_feedback` is tracked with `satisfied: true`.
- Confirm that the bottom sheet is then dismissed and a new web view is presented with a survey for more detailed feedback.
- Create another campaign and confirm that the feedback banner is available on the success bottom sheet.
- Tap on 👎  on the feedback banner, confirm that `blaze_campaign_creation_feedback` is tracked with `satisfied: false`.
- Confirm that the bottom sheet is then dismissed and a new web view is presented with a survey for more detailed feedback.

## Testing information

Tested on simulator iPhone 15 Pro iOS 17.4 for store with 0, 1, 2+ campaigns.

## Screenshots

With no campaign | With 1 or more campaign
--- | ---
 | 

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [x] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [x] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [x] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on all devices (phone/tablet) and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-ios/milestone/225">20.6</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false"><p>This PR contains changes to Tracks-related logic. Please ensure (<strong>author and reviewer</strong>) the following are completed:</p>

<ul>
  <li>The tracks events must be validated in the Tracks system.</li>
  <li>Verify the internal Tracks spreadsheet has also been updated.</li>
  <li>Please consider registering any new events.</li>
  <li>The PR must be assigned the <strong>category: tracks</strong> label.</li>
</ul></td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/Preview/ProductDetailPreviewView.swift [hunk 1]
```diff
@@ -223,13 +223,13 @@ private extension ProductDetailPreviewView {
     }
 
     var feedbackBanner: some View {
-        FeedbackView(title: Localization.feedbackQuestion,
+        FeedbackView(configuration: .init(title: Localization.feedbackQuestion,
```
Comment (selanthiraiyan) [inline WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/Preview/ProductDetailPreviewView.swift:226]: Super nit: Indentation needs to be updated for the following lines.

File: WooCommerce/Classes/Blaze/BlazeCampaignCreationCoordinator.swift [hunk 2]
```diff
@@ -267,18 +268,45 @@ private extension BlazeCampaignCreationCoordinator {
...
+        let feedbackConfiguration: FeedbackView.Configuration? = {
+            let allCampaigns = storageManager.viewStorage.loadAllBlazeCampaignListItems(siteID: siteID)
+            guard allCampaigns.isNotEmpty else {
+                /// Only ask for feedback if there is at least 1 campaign created in the past.
+                return nil
+            }
+            return FeedbackView.Configuration(title: Localization.feedbackQuestion, onVote: { [weak self] vote in
+                guard let self else { return }
+                analytics.track(event: .Blaze.campaignCreationFeedbackReceived(positive: vote == .up))
+                /// Add 0.3s delay before showing survey page
```
Comment (selanthiraiyan) [inline WooCommerce/Classes/Blaze/BlazeCampaignCreationCoordinator.swift]: nit: Could you update the comment to explain why we add this delay?

File: WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/Preview/LegacyProductDetailPreviewView.swift [hunk 3]
```diff
@@ -108,14 +108,14 @@ struct LegacyProductDetailPreviewView: View {
                 }
 
                 // Feedback banner
-                FeedbackView(title: Localization.feedbackQuestion,
+                FeedbackView(configuration: .init(title: Localization.feedbackQuestion,
```
Comment (selanthiraiyan) [inline WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/Preview/LegacyProductDetailPreviewView.swift:111]: Super nit: Indentation needs to be updated for the following lines.

File: WooCommerce/Classes/ViewRelated/Blaze/CampaignCreation/BlazeCampaignCreationFormViewModel.swift [hunk 4]
```diff
@@ -395,6 +395,7 @@ extension BlazeCampaignCreationFormViewModel {
 }
 
 private extension BlazeCampaignCreationFormViewModel {
+    @MainActor
```
Comment (selanthiraiyan) [inline WooCommerce/Classes/ViewRelated/Blaze/CampaignCreation/BlazeCampaignCreationFormViewModel.swift:398]: 👍

**Review by selanthiraiyan — COMMENTED** [review]
I validated that the survey was displayed after creating the second campaign. ✅ 

❓ I interacted with the survey popup and opened the webview, and next time I created a campaign, I saw the option for feedback once again. Is this expected? 



Do we want to avoid asking for feedback if the user has previously interacted with the thumbs up/down buttons?

File: WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/Preview/ProductDetailPreviewView.swift (hunk 1 above)
Comment (itsmeichigo) [reply WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/Preview/ProductDetailPreviewView.swift:226]: Fixed in 62be0c8d989412fc2af7fc1a896d7b5acaf0e6eb.

File: WooCommerce/Classes/Blaze/BlazeCampaignCreationCoordinator.swift (hunk 2 above)
Comment (itsmeichigo) [reply WooCommerce/Classes/Blaze/BlazeCampaignCreationCoordinator.swift]: Comment updated in f49c12ef4af54c7567fd0c63d1635cc68744c639.

File: WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/Preview/LegacyProductDetailPreviewView.swift (hunk 3 above)
Comment (itsmeichigo) [reply WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/Preview/LegacyProductDetailPreviewView.swift:111]: Done in 4f3768ea2ebcce9fd9b9bed52a309841c0a2731e.

**Comment by itsmeichigo** [comment]
Thanks @selanthiraiyan for the review!

> Do we want to avoid asking for feedback if the user has previously interacted with the thumbs up/down buttons?

@JorgeMucientes mentioned in the decision here pe5sF9-3di-p2 that he wants the banner to always be shown. I suppose we want users to keep sending feedback as they want.

**Review by selanthiraiyan — APPROVED** [review]
Thanks for the updates, Huong! 🚢

**Pull request merged**